ffi/diplomat/src/properties_maps.rs

```diff
@@ -0,0 +1,69 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use icu_properties::{
+        maps::{self, CodePointMapResult},
+        provider::UnicodePropertyMapV1Marker,
+        Script,
+    };
+    use icu_provider::prelude::DataPayload;
+
+    use crate::{provider::ffi::ICU4XDataProvider, provider::ffi::ICU4XStaticDataProvider};
+
+    #[diplomat::opaque]
+    /// An ICU4X Unicode Set Property object, capable of querying whether a code point is contained in a set based on a Unicode property.
+    /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/icu_properties/index.html) for more information.
+    pub struct ICU4XUnicodeScriptMapProperty(
+        DataPayload<'static, UnicodePropertyMapV1Marker<Script>>,
+    );
+
+    pub struct ICU4XUnicodeScriptMapPropertyResult {
+        /// The [`ICU4XUnicodeScriptMapProperty`], if creation was successful.
+        pub data: Option<Box<ICU4XUnicodeScriptMapProperty>>,
+        /// Whether creating the [`ICU4XUnicodeScriptMapProperty`] was successful.
+        pub success: bool,
```

Comment:
thought (nb): Tbh I feel like the `success` is redundant if `data` is an `Option`.

I eventually want to use `Result`s here but I'd like to fix some things in Diplomat first.

Reply:
I copied this from one of the other components (I think FixedDecimalFormat).  I do absolutely think we need to invest more in our `Result` story.  But I don't want to diverge from what we are doing elsewhere until we do it everywhere.